int64 utime
int32 num_particles
particle_t[] particles
